.travis.yml

```diff
@@ -1,13 +1,13 @@
 # .travis.yml - travis CI config for github
 
 sudo: required
-dist: xenial
+dist: bionic
 language: python
 python:
   - "3.6"
   - "3.7"
 before_install:
   - sudo apt-get -qq update
-  - sudo apt-get install -y python3-libtorrent
+  - pip install lbry-libtorrent
```

Comment:
I feel like this needs more discussion.  Is this change necessary, does `python3-libtorrent` not work under bionic anymore?  It's good that we upgrade to `bionic` for CI, but in general I use the OS packages for a reason -- that reason being that I intentionally depend on a slightly older version of the code.

In general I'm not using Travis much for CI anymore and I would probably switch this to GH Actions if I was doing it today.

Reply:
This is a big explanation and I have done some extensive testing. 

For now, suffice to say that
 python3-libtorrent via apt-get doesn't work in Travis at all for me, with both Xenial and bionic, and bionic used to work in some commits but it regressed, I suspect it is because it runs in an isolated virtualenv; 
that it likely won't work even if we use --system-site-packages in xenial (so Xenial is a no-go) (because it installs to the default python in the distro, 2.7.12 and 3.5.1 in Xenial)
That in my personal computer (arch and Manjaro) , if you run using the system python and you have libtorrent-rasterbar installed, it works. 
If you run inside a venv(system-site-packages off) , the only package that's able to communicatr with the system installed libtorrent, via pacman, is lbry-libtorrent.

Reply:
Hello Mr Banks, I found this to be a complicated issue, but I believe lbry-libtorrent is the best all-around solution for running within a venv.

I found out that Travis-CI will [use virtual environments](https://docs.travis-ci.com/user/languages/python/#travis-ci-uses-isolated-virtualenvs) if running tests in python, it will not use the system Python, so it is recommended to install via pip; and if you decide to use apt anyway you'll only be able to use the [default](https://docs.travis-ci.com/user/languages/python/#travis-ci-uses-isolated-virtualenvs) Python versions even if you specify `system-site-packages: True`, therefore the package will not be accessible to other python versions.

Default Python Versions
- Xenial: [2.7.12 and 3.5.1](https://docs.travis-ci.com/user/languages/python/#travis-ci-uses-isolated-virtualenvs);
- Bionic: [3.6](https://wiki.ubuntu.com/BionicBeaver/ReleaseNotes#Other_base_system_changes_since_16.04_LTS);
- FocalFossa: [3.8](https://wiki.ubuntu.com/FocalFossa/ReleaseNotes#Other_base_system_changes_since_18.04_LTS);

Therefore, using `system-site-packages: True`, should work in Xenial for python 3.5, and in Bionic for 3.6. The tests using python 3.7 should always fail. That is exactly what happens, python 3.6 and 3.7 fail in Xenial; Python 3.6 works in Bionic while 3.7 fails.

What is the best solution then?
If running under system python, just follow upstream libtorrent instructions.
If running in a venv, as Travis does, use a pip package. I could only find two working packages in pip: one that provides precompiled libtorrent binaries for x86_64, and one that provides a "bridge" between venv and system libtorrent. 
This package that provides a bridge is lbry-libtorrent.

Reply:
This could be moved to a separate issue, and we drop these specific changes, for organizational purposes, but right now, Travis-CI is broken.